### ISSUE TITLE
Possible bug with 101.02 - error "Soft limit - X min"

### ISSUE THREAD

**Issue by justinclift** [synthetos/g2#349]
Found what **seems** like a bug with 101.02 (current edge) today, when trying to cut out a ~reasonably simple circle, diamond, square test with machining wax.

![11 02-circle_diamond_square_v1-fusion360](https://user-images.githubusercontent.com/406299/40190843-bb6bc9ec-59f7-11e8-885c-d8a4869b2358.png)
![11 02-circle_diamond_square_v1-cncjs](https://user-images.githubusercontent.com/406299/40190850-bf6bae72-59f7-11e8-8499-042ea05f8858.png)

* With 11.02, the job errors with "Soft limit - X min" or "Soft limit - Z min"
  * Ran the job 3 times.  Failed 3 times (100% failure rate).
    * Twice gave the "Soft limit - Z min" error.  Once gave the "Soft limit - X" error.
* With 100.26 (current master), the job runs through to completion
  * Ran the job twice.  Both times ran from start to finish without issue.

Name|Value
---|---
Host computer|Windows 7 x64
G-Code sender|CNCjs 1.9.15 (Windows x64 desktop application version)
CNC hardware| Shapeoko 3, with g2core replacing the stock electronics/controller
g2core hardware| Ardunio Due with simple self built controller (4x A4988 stepper drivers)
g2core config| [settings_shapeoko3.h](https://github.com/justinclift/g2/blob/b9158edc2fd9f918f83a1f67a7d42f0d88763ba8/g2core/settings/settings_shapeoko3.h) (it's the shapeoko2 dual Y one, with [some minor adjustments](https://github.com/justinclift/g2/commit/b9158edc2fd9f918f83a1f67a7d42f0d88763ba8))
G-Code file| [circle diamond square - 10mm end mill.txt](https://github.com/synthetos/g2/files/2014034/circle.diamond.square.-.10mm.pass.txt)

This error seems especially weird, because as far as I can tell both soft and hard limits are disabled in my [settings config](https://github.com/justinclift/g2/blob/b9158edc2fd9f918f83a1f67a7d42f0d88763ba8/g2core/settings/settings_shapeoko3.h#L42-L43).

After the first run failed (using 11.02), I suspected it was potentially due to static electricity buildup or similar due to my crappy wiring/soldering.  So, reran things with the bench power supply not providing current to the motors.  Error still occured.  Not a static electricity problem after all.

It turned out to be a bit more complex to reproduce than the above sounds.  The machine needs to be physically homed first (eg with motors given power), then jogged into the starting position (X: 45, Y 385, Z: 4) then setting G54 0,0,0 at that point.  After this, power to the motors can be turned off as the error will still occur.

If the G-code file is sent to g2 without the above physical homing + G54 setup dance, then no error shows up. (this was a bit time consuming to work out)

Logs of the errors below
---
### Initial error log
Firmware 101.02 - commit 194c9ab

Machine starting co-ordinates unknown.  May not have been exactly X: 45.0, Y 385.0, Z: 4.0

I **think** the error occurred roughly 5 or 6 mins into the job.  Wasn't watching carefully enough at the time though.

```
...
{"sr":{"posz":1.24815,"vel":702.87,"dist":1,"stat":5}}
{"qr":48}
{"sr":{"posz":2,"vel":0,"dist":0,"stat":3}}
feeder> G91
{"r":{},"f":[1,0,4]}
feeder> G0 Z1
{"r":{},"f":[1,0,6]}
feeder> G90
{"qr":47}
{"r":{},"f":[1,0,4]}
{"sr":{"posz":2.24815,"vel":702.87,"dist":1,"stat":5}}
{"qr":48}
{"sr":{"posz":3,"vel":0,"dist":0,"stat":3}}
feeder> G91
{"r":{},"f":[1,0,4]}
feeder> G0 Z1
feeder> G90
{"r":{},"f":[1,0,6]}
{"qr":47}
{"r":{},"f":[1,0,4]}
{"sr":{"posz":3.40021,"vel":794.93,"dist":1,"stat":5}}
{"qr":48}
{"sr":{"posz":4,"vel":0,"dist":0,"stat":3}}
{"sr":{"stat":3}}
> X-1.653 Y-170.831 Z-4.347
{"err":{"code":225,"msg":"Soft limit - Z min","line":14366,"data":"X-1.653 Y-170.831 Z-4.347"}}
```
---

### Second error
Firmware 101.02 - commit 194c9ab

Machine starting co-ordinates (after homing) - X: 45.0, Y: 385.0, Z: 4.0

Error occurred roughly 5 min 15 seconds into the job.
```
...
feeder> G90
{"qr":47}
{"r":{},"f":[1,0,4]}
{"sr":{"posx":49.60943,"vel":1149.75,"dist":1,"stat":5}}
{"sr":{"posx":46.1574,"vel":2013.03}}
{"qr":48}
{"sr":{"posx":45,"vel":0,"dist":0,"stat":3}}
{"sr":{"stat":3}}
feeder> G10 L20 P1 X0
{"r":{},"f":[1,0,14]}
{"sr":{"posx":0,"stat":5}}
feeder> G10 L20 P1 Y0
{"r":{},"f":[1,0,14]}
{"sr":{"posy":0}}
feeder> G10 L20 P1 Z0
{"r":{},"f":[1,0,14]}
{"sr":{"posz":0}}
> X-1.786 Y-171.349 Z-7.898
{"err":{"code":225,"msg":"Soft limit - Z min","line":37618,"data":"X-1.786 Y-171.349 Z-7.898"}}
```

---
### Third error
Firmware 101.02 - commit 194c9ab

**Note** - The above CNCjs screenshot is from this run.

Machine starting co-ordinates (after homing) - X: 45.0, Y: 385.0, Z: 4.0

(Full log this time, as the error occured near the start of the job)
```
CNCjs 1.9.15 [TinyG]
Connected to COM5 with a baud rate of 115200
{"r":{"fv":0.99,"fb":101.03,"fbs":"101.02-17-gf08e","fbc":"settings_shapeoko3.h","hp":"ArduinoDue","hv":"na","id":"0084-d639-18c6-29c","msg":"SYSTEM READY"},"f":[1,0,1]}
feeder> {ej:1}
{"r":{"ej":1},"f":[1,0,8]}
feeder> {jv:4}
feeder> {qv:1}
{"r":{"jv":4},"f":[1,0,8]}
feeder> {sv:1}
feeder> {si:100}
{"r":{"qv":1},"f":[1,0,8]}
feeder> {spe:n}
{"r":{"sv":1},"f":[1,0,8]}
{"r":{"si":100},"f":[1,0,10]}
{"r":{"spe":null},"f":[1,100,9]}
{"err":{"code":100,"msg":"Unrecognized command or config name"}}
feeder> {sr:{line:t,vel:t,feed:t,stat:t,cycs:t,mots:t,hold:t,momo:t,coor:t,plan:t,unit:t,dist:t,frmo:t,path:t,posx:t,posy:t,posz:t,posa:t,posb:t,posc:t,mpox:t,mpoy:t,mpoz:t,mpoa:t,mpob:t,mpoc:t}}
{"r":{"sr":{"line":1,"vel":false,"feed":false,"stat":1,"cycs":1,"mots":1,"hold":1,"momo":1,"coor":1,"plan":1,"unit":1,"dist":1,"frmo":1,"path":1,"posx":false,"posy":false,"posz":false,"posa":false,"posb":false,"posc":false,"mpox":false,"mpoy":false,"mpoz":false,"mpoa":false,"mpob":false,"mpoc":false}},"f":[1,0,189]}
feeder> {sys:n}
{"r":{"sys":{"fb":101.03,"fv":0.99,"fbs":"101.02-17-gf08e","fbc":"settings_shapeoko3.h","hp":"ArduinoDue","hv":"na","id":"0084-d639-18c6-29c","jt":0.75,"ct":0.01,"zl":0,"sl":1,"lim":1,"saf":1,"m48":1,"froe":0,"fro":1,"troe":0,"tro":1,"mt":2,"tv":1,"ej":1,"jv":4,"qv":1,"sv":1,"si":100,"gpl":0,"gun":1,"gco":1,"gpa":2,"gdi":0}},"f":[1,0,9]}
feeder> {mt:n}
{"r":{"mt":2},"f":[1,0,8]}
feeder> {pwr:n}
{"r":{"pwr":{"1":0,"2":0,"3":0,"4":0}},"f":[1,0,9]}
feeder> {qr:n}
{"r":{"qr":48},"f":[1,0,8]}
feeder> {sr:n}
{"r":{"sr":{"line":0,"posx":0,"posy":0,"posz":0,"posa":0,"posb":0,"posc":0,"feed":0,"vel":0,"unit":1,"coor":1,"dist":0,"admo":1,"frmo":1,"momo":4,"stat":1}},"f":[1,0,8]}
client> G28.2 X0 Y0 Z0
{"r":{},"f":[1,0,15]}
{"qr":47}
{"qr":45}
{"qr":46}
{"sr":{"posz":0.20359,"feed":1200,"vel":553.66,"coor":0,"dist":1,"momo":1,"stat":9}}
{"sr":{"posz":1.95166,"vel":1200}}
{"sr":{"posz":3.96156}}
{"sr":{"posz":5.97147}}
{"sr":{"posz":7.98137}}
{"sr":{"posz":9.99127}}
{"sr":{"posz":12.00117}}
{"sr":{"posz":14.01107}}
{"sr":{"posz":16.02098}}
{"sr":{"posz":18.03088}}
{"sr":{"posz":20.04078}}
{"sr":{"posz":22.05068}}
{"sr":{"posz":24.06058}}
{"sr":{"posz":26.07048}}
{"sr":{"posz":28.08039}}
{"sr":{"posz":30.06029}}
{"sr":{"posz":32.07019}}
{"sr":{"posz":34.08009}}
{"sr":{"posz":36.09}}
{"sr":{"posz":38.09989}}
{"sr":{"posz":40.1098}}
{"sr":{"posz":42.1197}}
{"sr":{"posz":44.1296}}
{"sr":{"posz":46.1395}}
{"sr":{"posz":48.14941}}
{"sr":{"posz":50.12931}}
{"sr":{"posz":52.13921}}
{"sr":{"posz":54.14911}}
{"sr":{"posz":56.15901}}
{"sr":{"posz":58.16892}}
{"sr":{"posz":60.17882}}
{"sr":{"posz":62.18872}}
{"sr":{"posz":64.19863}}
{"sr":{"posz":66.20853}}
{"sr":{"posz":67.8504,"vel":417.77}}
{"qr":47}
{"qr":46}
{"sr":{"posz":67.95391,"vel":141.09}}
{"sr":{"posz":66.6574,"vel":1198.88}}
{"sr":{"posz":64.67533,"vel":1058.91}}
{"sr":{"posz":63.98022,"vel":1.12}}
{"qr":47}
{"qr":48}
{"qr":46}
{"sr":{"posz":64.09985,"feed":100,"vel":100}}
{"sr":{"posz":64.26477}}
{"sr":{"posz":64.43219}}
{"sr":{"posz":64.59961}}
{"sr":{"posz":64.76703}}
{"sr":{"posz":64.93196}}
{"sr":{"posz":65.09938}}
{"sr":{"posz":65.26679}}
{"sr":{"posz":65.43421}}
{"sr":{"posz":65.60163}}
{"sr":{"posz":65.76905}}
{"sr":{"posz":65.93648}}
{"sr":{"posz":66.1039}}
{"sr":{"posz":66.27132}}
{"sr":{"posz":66.43874}}
{"qr":47}
{"qr":46}
{"sr":{"posz":66.43007,"feed":1200,"vel":245.39}}
{"sr":{"posz":65.18825,"vel":917.35}}
{"sr":{"posz":64.48853,"vel":3.61}}
{"qr":47}
{"qr":48}
{"qr":46}
{"sr":{"posx":1.20094,"posz":68,"feed":2000,"vel":2000}}
{"sr":{"posx":4.55092}}
{"sr":{"posx":7.90091}}
{"sr":{"posx":11.25089}}
{"sr":{"posx":14.60088}}
{"sr":{"posx":17.95086}}
{"sr":{"posx":21.30085}}
{"sr":{"posx":24.65083}}
{"sr":{"posx":28.00082}}
{"sr":{"posx":31.3508}}
{"sr":{"posx":34.65079}}
{"sr":{"posx":38.00077}}
{"sr":{"posx":41.35076}}
{"sr":{"posx":44.70074}}
{"sr":{"posx":48.05073}}
{"sr":{"posx":51.40071}}
{"sr":{"posx":54.7507}}
{"sr":{"posx":58.10068}}
{"sr":{"posx":61.45067}}
{"sr":{"posx":64.75065}}
{"sr":{"posx":68.10064}}
{"sr":{"posx":71.45063}}
{"sr":{"posx":74.80061}}
{"sr":{"posx":78.15059}}
{"sr":{"posx":81.50058}}
{"sr":{"posx":84.85057}}
{"sr":{"posx":88.20055}}
{"sr":{"posx":91.55053}}
{"sr":{"posx":94.85052}}
{"sr":{"posx":98.2005}}
{"sr":{"posx":101.55049}}
{"sr":{"posx":104.90048}}
{"sr":{"posx":108.25046}}
{"sr":{"posx":111.60044}}
{"sr":{"posx":114.95043}}
{"sr":{"posx":118.30042}}
{"sr":{"posx":121.6504}}
{"sr":{"posx":124.95039}}
{"sr":{"posx":128.30036}}
{"sr":{"posx":131.65036}}
{"sr":{"posx":135.00033}}
{"sr":{"posx":138.35032}}
{"sr":{"posx":141.7003}}
{"sr":{"posx":145.05029}}
{"sr":{"posx":148.40028}}
{"sr":{"posx":151.75025}}
{"sr":{"posx":155.05024}}
{"sr":{"posx":158.40023}}
{"sr":{"posx":161.75021}}
{"sr":{"posx":165.1002}}
{"sr":{"posx":168.45019}}
{"sr":{"posx":171.80017}}
{"sr":{"posx":175.15016}}
{"sr":{"posx":178.50015}}
{"sr":{"posx":181.80012}}
{"sr":{"posx":185.15011}}
{"sr":{"posx":188.5001}}
{"sr":{"posx":191.85008}}
{"sr":{"posx":195.20007}}
{"sr":{"posx":198.55004}}
{"sr":{"posx":201.90003}}
{"sr":{"posx":205.25003}}
{"sr":{"posx":208.6}}
{"sr":{"posx":211.89999}}
{"sr":{"posx":215.24998}}
{"sr":{"posx":218.59996}}
{"sr":{"posx":221.94995}}
{"sr":{"posx":225.29994}}
{"sr":{"posx":228.64991}}
{"sr":{"posx":231.9999}}
{"sr":{"posx":235.34989}}
{"sr":{"posx":238.69987}}
{"sr":{"posx":241.99986}}
{"sr":{"posx":245.34985}}
{"sr":{"posx":248.69982}}
{"sr":{"posx":252.04981}}
{"sr":{"posx":255.39979}}
{"sr":{"posx":258.74978}}
{"sr":{"posx":262.09976}}
{"sr":{"posx":265.44976}}
{"sr":{"posx":268.79974}}
{"sr":{"posx":272.09973}}
{"sr":{"posx":275.4497}}
{"sr":{"posx":278.79971}}
{"sr":{"posx":282.14968}}
{"sr":{"posx":285.49966}}
{"sr":{"posx":288.84963}}
{"sr":{"posx":292.19964}}
{"sr":{"posx":295.54962}}
{"sr":{"posx":298.89959}}
{"sr":{"posx":302.19958}}
{"sr":{"posx":305.54959}}
{"sr":{"posx":308.89956}}
{"sr":{"posx":312.24954}}
{"sr":{"posx":315.59954}}
{"sr":{"posx":318.94952}}
{"sr":{"posx":322.29949}}
{"sr":{"posx":325.64947}}
{"sr":{"posx":328.94946}}
{"sr":{"posx":332.29946}}
{"sr":{"posx":335.64944}}
{"sr":{"posx":338.99942}}
{"sr":{"posx":342.34942}}
{"sr":{"posx":345.6994}}
{"sr":{"posx":349.04937}}
{"sr":{"posx":352.39938}}
{"sr":{"posx":355.74935}}
{"sr":{"posx":359.04934}}
{"sr":{"posx":362.39932}}
{"sr":{"posx":365.74929}}
{"sr":{"posx":369.0993}}
{"sr":{"posx":372.44927}}
{"sr":{"posx":375.79925}}
{"qr":47}
{"qr":46}
{"sr":{"posx":377.31268,"vel":1000}}
{"sr":{"posx":374.10574,"vel":1992.26}}
{"qr":47}
{"qr":48}
{"qr":46}
{"sr":{"posx":373.53414,"feed":100,"vel":100}}
{"sr":{"posx":373.70162}}
{"sr":{"posx":373.86911}}
{"sr":{"posx":374.03659}}
{"sr":{"posx":374.20407}}
{"sr":{"posx":374.37155}}
{"sr":{"posx":374.53903}}
{"sr":{"posx":374.70651}}
{"sr":{"posx":374.87399}}
{"sr":{"posx":375.04147}}
{"sr":{"posx":375.20645}}
{"sr":{"posx":375.37396}}
{"sr":{"posx":375.54144}}
{"sr":{"posx":375.70892}}
{"sr":{"posx":375.8764}}
{"sr":{"posx":376.04388}}
{"sr":{"posx":376.21136}}
{"sr":{"posx":376.37884}}
{"sr":{"posx":376.54632}}
{"qr":47}
{"qr":46}
{"sr":{"posx":374.62817,"feed":2000,"vel":774.94}}
{"qr":47}
{"qr":48}
{"qr":46}
{"sr":{"posx":420,"posy":1.77405,"vel":2000}}
{"sr":{"posy":5.12368}}
{"sr":{"posy":8.47331}}
{"sr":{"posy":11.82294}}
{"sr":{"posy":15.17257}}
{"sr":{"posy":18.5222}}
{"sr":{"posy":21.87183}}
{"sr":{"posy":25.22146}}
{"sr":{"posy":28.5211}}
{"sr":{"posy":31.87073}}
{"sr":{"posy":35.22035}}
{"sr":{"posy":38.56999}}
{"sr":{"posy":41.91962}}
{"sr":{"posy":45.26924}}
{"sr":{"posy":48.61888}}
{"sr":{"posy":51.96651,"vel":1959.09}}
{"qr":47}
{"qr":46}
{"sr":{"posy":51.47357,"vel":2000}}
{"sr":{"posy":48.55863,"vel":294.84}}
{"qr":47}
{"qr":48}
{"qr":46}
{"sr":{"posy":48.67687,"feed":100,"vel":100}}
{"sr":{"posy":48.84433}}
{"sr":{"posy":49.01179}}
{"sr":{"posy":49.17926}}
{"sr":{"posy":49.34672}}
{"sr":{"posy":49.51418}}
{"sr":{"posy":49.68164}}
{"sr":{"posy":49.8491}}
{"sr":{"posy":50.01406}}
{"sr":{"posy":50.18152}}
{"sr":{"posy":50.34898}}
{"sr":{"posy":50.51644}}
{"sr":{"posy":50.68391}}
{"sr":{"posy":50.85137}}
{"sr":{"posy":51.01882}}
{"sr":{"posy":51.18629}}
{"sr":{"posy":51.35125}}
{"sr":{"posy":51.51871}}
{"qr":47}
{"qr":46}
{"sr":{"posy":51.428,"feed":2000,"vel":1225.06}}
{"qr":47}
{"qr":48}
{"qr":47}
{"qr":48}
{"sr":{"posy":435,"feed":0,"vel":0,"coor":1,"dist":0,"momo":4,"stat":3}}
{"sr":{"stat":3}}
> G0 X45 Y385 Z4
{"r":{},"f":[1,0,15]}
{"qr":47}
{"sr":{"posx":419.56649,"posy":434.94219,"posz":67.92602,"vel":1311.89,"momo":0,"stat":5}}
{"sr":{"posx":412.87533,"posy":434.05004,"posz":66.78406,"vel":6571.89}}
{"sr":{"posx":401.44491,"posy":432.50259,"posz":64.80328,"vel":7194.26}}
{"sr":{"posx":389.67236,"posy":430.95632,"posz":62.82408}}
{"sr":{"posx":377.89981,"posy":429.38665,"posz":60.8149}}
{"sr":{"posx":366.12725,"posy":427.81698,"posz":58.80572}}
{"sr":{"posx":354.53039,"posy":426.27075,"posz":56.82652}}
{"sr":{"posx":342.75784,"posy":424.70108,"posz":54.81734}}
{"sr":{"posx":330.98529,"posy":423.1314,"posz":52.80816}}
{"sr":{"posx":319.21273,"posy":421.5617,"posz":50.79898}}
{"sr":{"posx":307.44018,"posy":419.99203,"posz":48.78979}}
{"sr":{"posx":295.84332,"posy":418.4458,"posz":46.8106}}
{"sr":{"posx":284.07077,"posy":416.87612,"posz":44.80142}}
{"sr":{"posx":272.29821,"posy":415.30645,"posz":42.79224}}
{"sr":{"posx":260.52566,"posy":413.73678,"posz":40.78305}}
{"sr":{"posx":248.75311,"posy":412.16711,"posz":38.77387}}
{"sr":{"posx":236.98056,"posy":410.59744,"posz":36.76469}}
{"sr":{"posx":225.38371,"posy":409.05117,"posz":34.78549}}
{"sr":{"posx":213.61116,"posy":407.4815,"posz":32.77631}}
{"sr":{"posx":201.8386,"posy":405.91183,"posz":30.76713}}
{"sr":{"posx":190.06605,"posy":404.34216,"posz":28.75794}}
{"sr":{"posx":178.2935,"posy":402.77249,"posz":26.74876}}
{"sr":{"posx":166.52095,"posy":401.20281,"posz":24.73958}}
{"sr":{"posx":154.9241,"posy":399.65655,"posz":22.76039}}
{"sr":{"posx":143.15155,"posy":398.08688,"posz":20.7512}}
{"sr":{"posx":131.37899,"posy":396.51721,"posz":18.74202}}
{"sr":{"posx":119.60644,"posy":394.94754,"posz":16.73284}}
{"sr":{"posx":107.83389,"posy":393.37786,"posz":14.72365}}
{"sr":{"posx":96.06134,"posy":391.80819,"posz":12.71447}}
{"sr":{"posx":84.46449,"posy":390.26196,"posz":10.73528}}
{"sr":{"posx":72.69194,"posy":388.69229,"posz":8.72609}}
{"sr":{"posx":60.91938,"posy":387.12261,"posz":6.71691}}
{"sr":{"posx":49.73348,"posy":385.6311,"posz":4.80785,"vel":5672.33}}
{"sr":{"posx":45.0953,"posy":385.01269,"posz":4.01627,"vel":491.06}}
{"qr":48}
{"sr":{"posx":45,"posy":385,"posz":4,"vel":0,"stat":3}}
{"sr":{"stat":3}}
> G0 X-2.43 Y-173.863
{"err":{"code":221,"msg":"Soft limit - X min","line":13,"data":"G0 X-2.43 Y-173.863"}}
>
```

**Comment by amx1** [comment]
@justinclift 

I am just checking the test cases. This one runs also without errors with my config and my custom shield.

Firmware: 101.02-17-g51f9-dirty, CNCJS 1.9.15 latest master, Soft and Hard limits enabled! Only needed digital inputs enabled to avoid random interrupts

**Comment by justinclift** [comment]
That sounds interesting. :smile:

When you say "Only needed digital inputs enabled to avoid random interrupts", does that mean I should change something in [my config](https://github.com/justinclift/g2/blob/b9158edc2fd9f918f83a1f67a7d42f0d88763ba8/g2core/settings/settings_shapeoko3.h)?

**Comment by amx1** [comment]
check what happens if you set all not used inputs to disabled
like #define DI1_MODE  IO_MODE_DISABLED

if your custom shield does not set the inputs to a defined value by pull-up resistors this might fire random interrupts

**Comment by justinclift** [comment]
Oh interesting thought.  Thanks @amx1 I'll give that a shot. :smile:

**Comment by aldenhart** [comment]
Justin, did that solve the issue?

**Comment by justinclift** [comment]
Haven't checked yet.  Been head down migrating servers for a different project + in-depth reviewing some of the PR's there too.

Likely be finished with those bits over the weekend, which should free up some schedule. :smile:

**Comment by aldenhart** [comment]
Cool. I'm interested in your experience.

**Comment by justinclift** [comment]
@aldenhart Just tried a run of this with latest edge (101.03) with the motors disabled, with the [same settings](https://github.com/justinclift/g2/commit/ca4e95de8d589dfd22ca7b376d2d5685df73559e) applied.

It ran through to completion without error, so I'm kind of hopeful your fix the other day (5c4ba36db4a2746a27f5c0b03cb5220497a1a9a5) solved the problem I was hitting.

I'll try it again tommorrow morning with the motors enabled and actually cutting something, and report back. :smile:

**Comment by justinclift** [comment]
Yep, this issue seems to be fixed in 101.03. :smile:

With 101.02, this particular gcode file had a 100% failure rate.  Now, after a few soft runs (eg motors off) and one hard run (cutting stuff) it's not yet failed.

![edge-101 03-circle_diamond_square - 1](https://user-images.githubusercontent.com/406299/40977774-e1a5352c-68c9-11e8-8a57-3d8ac877a4e7.jpg)

Thanks @aldenhart and @amx1. :smile:
